### PR TITLE
Aplpy ruff rules (RUF)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**

### CODE REVIEW

File: xarray/__init__.py [hunk 1]
```diff
@@ -65,10 +65,27 @@
...
+    "IndexSelResult",
+    "IndexVariable",
+    # Exceptions
+    "InvalidTreeError",
+    "MergeError",
+    "NamedArray",
+    "NotFoundInTreeError",
+    "SerializationWarning",
+    "TreeIsomorphismError",
+    "Variable",
```
Comment (headtr1ck) [inline xarray/__init__.py:86]: The sorting of this list is wrong now. `Variable` is not an Error.

**Review by headtr1ck — COMMENTED** [review]
Add this rule to `pyproject.toml`.

Also, what is wrong with micromamba?

File: xarray/__init__.py (hunk 1 above)
Comment (keewis) [reply xarray/__init__.py:86]: yeah, I don't think we want to alphabetically sort that list if we want to keep the sections

Comment (dcherian) [reply xarray/__init__.py:86]: Yes, let's keep this list as-is

Comment (max-sixty) [reply xarray/__init__.py:86]: @DimitriPapadopoulos if you want to pass feedback upstream — the sorting should split on comments, so Sections work

Comment (DimitriPapadopoulos) [reply xarray/__init__.py:86]: It feels like a very specialised need. In this case, I will revert and add a `noqa`.

Comment (DimitriPapadopoulos) [reply xarray/__init__.py]: Actually, `noqa` is not needed yet. This is a "preview" rule.

Comment (max-sixty) [reply xarray/__init__.py]: > It feels like a very specialised need

FWIW I think this is fairly common in code formatters. I think rust imports are done this way, for example...

**Pull request merged**